File: Contest/3contest/HotDays.py
import math;

s = input()
a = s.find(" ")
n = int(s[0:a])
students = int(s[a+1:len(s)])

sum = 0

for i in range(n):
    s = input()
    a = s.find(" ")
    temp = int(s[0:a])
    s = s[a+1:len(s)]

    a = s.find(" ")
    maxTemp = int(s[0:a])
    s = s[a+1:len(s)]

    a = s.find(" ")
    studentsCost = int(s[0:a])
    s = s[a+1:len(s)]

    busCost = int(s)

    if temp + students < maxTemp:
        oneBus = busCost
    else:
        oneBus = busCost + students * studentsCost
    if temp < maxTemp:
        diffTemp = maxTemp - temp
        nBuses = math.ceil(students / diffTemp)
        if nBuses != 0:
            busesCost = nBuses * busCost
        else:
            busesCost = oneBus
    else:
        busesCost = oneBus
    sum += min(oneBus, busesCost)

print(sum)

File: Gym/3contest/HotDays.cpp
#include <iostream>
#include <algorithm>
#include <cmath>

using namespace std;

int main(){
    int n, students, temp, maxTemp, studentsCost, busCost;
    int min, oneBus, diffTemp, nBuses, busesCost;
    int sum;
    cin >> n >> students;
    while(n--){
        oneBus = 0;
        busesCost = 0;
        cin >> temp >> maxTemp >> studentsCost >> busCost;
        if(temp + students < maxTemp)oneBus = busCost;
        else oneBus = students * studentsCost + busCost;
        if(temp < maxTemp){
            diffTemp = maxTemp - temp;
            nBuses = ceil((double)students / diffTemp);
            if(nBuses != 0) busesCost = nBuses * busCost;
            else busesCost = oneBus;
        }else busesCost = oneBus;
        min = std::min(oneBus, busesCost);
        sum += min;
    }
    cout << sum << endl;
}
